File: HW 8 (Networking) — копия/multithreading/src/main/java/com/example/multithreading/database/ContactDao.java
package com.example.multithreading.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;
import static androidx.room.OnConflictStrategy.REPLACE;

@Dao
public interface ContactDao {

    @Query("SELECT * FROM CONTACT_TABLE")
    List<Contact> getAll();

    @Query("SELECT * FROM CONTACT_TABLE WHERE id LIKE :id")
    Contact getById(String id);

    @Insert(onConflict = REPLACE)
    void insert(Contact contact);

    @Update
    void update(Contact contact);

    @Delete
    void delete(Contact contact);
}

File: HW 9 (ContentProvider, ContentResolver)/contentprovider/src/main/java/com/example/contentprovider/database/ContactDao.kt
package com.example.contentprovider.database

import android.database.Cursor
import androidx.room.*
import androidx.room.OnConflictStrategy.REPLACE

@Dao
interface ContactDao{

    @Query("SELECT * FROM CONTACT_TABLE")
    fun getAll(): List<Contact>

    @Query("SELECT * FROM CONTACT_TABLE")
    fun getAllCursor(): Cursor

    @Query("SELECT * FROM CONTACT_TABLE WHERE id LIKE :id")
    fun getById(id: String): Contact

    @Query("SELECT * FROM CONTACT_TABLE WHERE id LIKE :id")
    fun getByIdCursor(id: String): Cursor

    @Insert(onConflict = REPLACE)
    fun insert(contact: Contact): Int

    @Update
    fun update(contact: Contact): Int

    @Delete
    fun delete(contact: Contact): Int

    @Delete
    fun delete(uriId: Long): Int
}